src/test/java/api/support/fakes/FakeOkapi.java

```diff
@@ -335,6 +335,13 @@ public void start(Promise<Void> startFuture) throws IOException {
       .create()
       .register(router);
 
+    new FakeStorageModuleBuilder()
+      .withRecordName("checkIns")
+      .withRootPath("/check-in-storage/check-ins")
+      .withCollectionPropertyName("checkIns")
+      .create()
+      .register(router);
```

Comment:
Is it worth validating the schema, given that the system is writing records using this API?

Reply:
I'm not sure if we have to add schema for all the endpoints (maybe for some "main" endpoints only, like loan, item, requests), because it introduces additional dependencies.

Reply:
> I'm not sure if we have to add schema for all the endpoints (maybe for some "main" endpoints only, like loan, item, requests), because it introduces additional dependencies.

What dependency does this introduce?

> What do you think?

Personally, I think that the tests should validate the schema of records that are written in the production code.

I think this relates to the question of whether failing to record a check in should fail the check in process itself.

Reply:
> What dependency does this introduce?

We have to maintain the copy of schema in unit tests each time it has been changed.

Reply:
I think the dependency on an expected version of the API is already present.

Personally, I think checking that the records that are written by the system comply with expectations is important (and I should have advocated for this in other pull requests)